server/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -228,29 +230,29 @@ public static ExecutionPlan planCopyFromExecution(AnalyzedCopyFrom copyFrom,
             .map(table::getReference)
             .collect(Collectors.toList());
 
-        List<Symbol> toCollect = getSymbolsRequiredForShardIdCalc(
-            primaryKeyRefs,
-            table.partitionedByColumns(),
-            clusteredBy == null ? null : table.getReference(clusteredBy)
-        );
-        Reference rawOrDoc = rawOrDoc(table, partitionIdent);
-        final int rawOrDocIdx = toCollect.size();
-        toCollect.add(rawOrDoc);
-
-        String[] excludes = partitionedByNames.size() > 0
-            ? partitionedByNames.toArray(new String[0]) : null;
-
-        InputColumns.SourceSymbols sourceSymbols = new InputColumns.SourceSymbols(toCollect);
-        Symbol clusteredByInputCol = null;
-        if (clusteredBy != null) {
-            clusteredByInputCol = InputColumns.create(table.getReference(clusteredBy), sourceSymbols);
-        }
-
-        SourceIndexWriterProjection sourceIndexWriterProjection;
+        List<Symbol> toCollect;
+        AbstractIndexWriterProjection indexWriterProjection;
         List<? extends Symbol> projectionOutputs = AbstractIndexWriterProjection.OUTPUTS;
         boolean returnSummary = copyFrom instanceof AnalyzedCopyFromReturnSummary;
         boolean failFast = boundedCopyFrom.settings().getAsBoolean("fail_fast", false);
+
         if (returnSummary || failFast) {
+
+            toCollect = getSymbolsRequiredForShardIdCalc(
+                primaryKeyRefs,
+                table.partitionedByColumns(),
+                clusteredBy == null ? null : table.getReference(clusteredBy)
+            );
+            Reference rawOrDoc = rawOrDoc(table, partitionIdent);
+            final int rawOrDocIdx = toCollect.size();
+            toCollect.add(rawOrDoc);
+
+            InputColumns.SourceSymbols sourceSymbols = new InputColumns.SourceSymbols(toCollect);
+            Symbol clusteredByInputCol = null;
+            if (clusteredBy != null) {
+                clusteredByInputCol = InputColumns.create(table.getReference(clusteredBy), sourceSymbols);
```

Comment:
I kept _raw related code for RETURN SUMMARY to keep the scope smaller. RETRN SUMMARY will be done in a follow up and this code will be gone.